cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/common/MetricStructuredTable.java

```diff
@@ -189,6 +189,12 @@ public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, String o
     return scan(() -> structuredTable.scan(keyRange, limit, orderByField, sortOrder), "sort.index.range.scan.");
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> scan(Range keyRange, int limit, Field<?> filterIndex, SortOrder sortOrder)
+    throws InvalidFieldException, IOException {
+    return scan(() -> structuredTable.scan(keyRange, limit, filterIndex, sortOrder), "pk.sort.index.range.scan.");
```

Comment:
I have no idea what to name this. This is to sort on pk and filter on index @sanjanasandeep

Reply:
It should be scan? To keep it consistent with the others in the spi

Reply:
I mean that string part, `"pk.sort.index.range.scan."`

Reply:
yeah sounds good , maybe "sort.primarykey.index.range.scan."

Reply:
Why pk? I don't think filter is limited to primary key

Reply:
The current naming is weird though. It seems that filtering should be indicated by "index", but `scan(Range keyRange, int limit, String orderByField, SortOrder sortOrder)` already uses `sort.index.range.scan.`. I'd fix it to be `sort.range.scan.` and make new method (this one) be `sort.index.range.scan.` It would be consistent

Reply:
The current naming is weird though. It seems that filtering should be indicated by "index", but `scan(Range keyRange, int limit, String orderByField, SortOrder sortOrder)` already uses `sort.index.range.scan.`. I'd fix it to be `sort.range.scan.` and make new method (this one) be `sort.index.range.scan.` It would be consistent

Reply:
because it's filtering on index but sorting on primary key

Reply:
I am looking on other methods they already provide sorting, but don't use "pk". Overall all this class is very strange. Such attribution should be done with tags.